src/cli/commands/object/put.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const Command = require('ronin').Command
+const utils = require('../../utils')
+const bs58 = require('bs58')
+const bl = require('bl')
+const fs = require('fs')
+const mDAG = require('ipfs-merkle-dag')
+const DAGNode = mDAG.DAGNode
+const debug = require('debug')
+const log = debug('cli:object')
+log.error = debug('cli:object:error')
+
+module.exports = Command.extend({
+  desc: 'Stores input as a DAG object, outputs its key',
+
+  options: {},
+
+  run: (filePath) => {
+    var ipfs = utils.getIPFS()
+
+    function parseAndAddNode (buf) {
+      if (utils.isDaemonOn()) {
+        return ipfs.object.put(buf, 'json', (err, obj) => {
+          if (err) {
+            log.error(err)
+            throw err
+          }
+
+          console.log('added', obj.Hash)
+        })
+      }
+
+      let parsed
+      try {
+        parsed = JSON.parse(buf.toString())
+        parsed.Links = parsed.Links || []
+      } catch (err) {
+        log.error(err)
+        throw new Error('failed to parse JSON: ' + err)
+      }
+
+      const data = new Buffer(parsed.Data)
+      const links = parsed.Links.map((link) => ({
+        name: link.Name,
+        hash: new Buffer(bs58.decode(link.Hash)),
+        size: link.Size
+      }))
+
+      const dagNode = new DAGNode(data, links)
+
+      ipfs.object.put(dagNode, (err, obj) => {
+        if (err) {
+          log.error(err)
+          throw err
+        }
+
+        console.log('added', bs58.encode(dagNode.multihash()).toString())
+      })
+    }
+
+    if (filePath) {
+      return parseAndAddNode(fs.readFileSync(filePath))
+    }
+
+    process.stdin.pipe(bl((err, input) => {
+      if (err) {
+        log.error(err)
+        throw err
+      }
+
+      parseAndAddNode(input)
+    }))
```

Comment:
`go-ipfs` doesn't prevent it either, if you pass a file path it simply takes priority over the `stdin` :)